THC-1366: adding api documentation for validate and optimize j1ql endpoints (#829)
JupiterOne/docs: knowledgeBase/APIs/jupiterone-api.md

}
```

## Validate and Optimize Queries

> You can use the validate and optimize j1ql endpoints to validate and/or optimize your queries.

Author: Let me know if this description can be better or isn't necessary

### Validate queries

The validate endpoint takes in an array of queries sent as a json object in the request body. It returns an array of objects containing the query and a boolean flag indicating whether the query is valid or not.

```text
POST /j1ql/validate
```

```json
{
    "queries": [
        "Find Person as p that relates to User"
    ]
}
```

#### Sample response

```json
[
    {
        "query": "Find Person as p that relates to User",
        "valid": true
    }
]
```

### Optimize queries

> The optimize endpoint optimizes and validates a query. It takes in an array of queries and return an array of objects containing the original query, the optimized query with auto-applied optimizations, the list of optimizations that were auto-applied and suggested optimizations that the user can apply, and a valid boolean flag indicating whether the query is valid or not.

Reviewer: ```suggestion
The optimize endpoint validates and optimizes a query. It takes in an array of queries and return an array of objects containing the original query, the optimized query, and a list of optimizations that are suggested, and a valid boolean flag indicating whether the query is valid or not.  
```

I don't think the caller needs to worry about the distinction between auto-applied and suggested for this endpoint.  It'll be their job to decide what to do with the optimized query.

Author: Ah yep that makes sense to me. committed your suggestion, thank you!
